src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -124,33 +125,32 @@ private OverdueFineCalculatorService(Clients clients) {
 
   private CompletableFuture<Result<CalculationParameters>> lookupItemRelatedRecords(
     CalculationParameters params) {
+
+    if (params.feeFine == null) {
+      return completedFuture(succeeded(params));
+    }
```

Comment:
What is the `feeFine` parameter?

What does it mean when it is null?

Reply:
`feeFine` is a description of a fee/fine type. It defines the name of a fine, its default amount, the owner etc. Think of it as a class of a fee/fine, and an "account" (yes, dumb name) is an instance of that class.

If it is null, it means it is missing from the DB, and this is not supposed to happen to automatic fees/fines. But still, it is not enough to fail the whole process, say a check-in. I wonder if we should log an error message here...

Reply:
> `feeFine` is a description of a fee/fine type. It defines the name of a fine, its default amount, the owner etc. Think of it as a class of a fee/fine, and an "account" (yes, dumb name) is an instance of that class.

Does that mean the `feeFine` parameter is the name of the fee fine type to search for?

> If it is null, it means it is missing from the DB, and this is not supposed to happen to automatic fees/fines. But still, it is not enough to fail the whole process, say a check-in. I wonder if we should log an error message here...

I tend to have a rather different view on this than other folks. Ultimately, I think this is a question of process design and needs to go back to the PO for an answer.

I think systems should actively fail when they encounter situations like this because they prompt folks to resolve the configuration issue before proceeding.

Otherwise, we find ourselves in situations where many fines aren't issued because the type wasn't set up or it was deleted. This is harder to recover from than a single operation failing.

My concern with log messages is that they need to be observed to be acted upon. Who monitors them? And how do they do that? I don't think FOLIO has clear answers to these questions, and so I think most log messages will go unobserved.

These are only my thoughts, not an official FOLIO position or instruction.

Does that make sense?

Reply:
> Does that mean the feeFine parameter is the name of the fee fine type to search for?

No, this is an actual fee/fine retrieved from the DB by name, which is hardcoded.

Reply:
> > Does that mean the feeFine parameter is the name of the fee fine type to search for?
> 
> No, this is an actual fee/fine retrieved from the DB by name, which is hardcoded.

What do you mean by an actual fee/fine, is that a fine itself or a type of fee / fine?

> You are raising a valid concern. Currently, there is no existing way to notify a user that a fee/fine type was not found in the DB. Even if we return an error from check-in process, there is no handler in the UI that will show an error message to the user. There will be a separate story in `mod-feesfines` to handle this case.

Ok. If the intention is to keep the silent failure (from the users perspective), I think this needs documenting in the readme and the 2020 Q1 release notes.

Reply:
Sure, we will document it.